### DISCUSSION TITLE
Updating Page Data With Data Refresh

### DISCUSSION THREAD

**Discussion by Zibbp** [nuxt/nuxt#17357] [Questions]
Hi,
I'm trying to update the data on the page and within components when I call the refresh function from `useAsyncData`. This is how I'm fetching the data.
```
const { data: queue, refresh } = await useAsyncData(
  `queue-${route.params.id}`,
  () => $fetch(`${config.apiURL}/api/v1/queue/${route.params.id}`)
);
```
I am then using a setInterval to refresh the data every second.
```
onMounted(async () => {
  intervalId = window.setInterval(async () => {
    refresh();
  }, 1000);
});
```
The HTTP request goes through and I can see updated data in the console network tab, but I cannot get the data on the page to refresh. I've tried invalidating the cache using ```refreshNuxtData(`queue-${route.params.id}`);``` but it still will not refresh the data on the page.

The only way I've got it working is by re-rendering the component. Is there something I am missing when calling the `refresh` function to update page data?

**Comment by Flanker72** [comment]
Maybe it's not your case, but I have [the similar issue](https://github.com/nuxt/framework/discussions/5418).
If you multiple instances of `useAsyncData`, then `refresh` function will trigger only its data reload, not all the instances with the same key.
So, if you have a page which displays data from `useAsyncData` instance, and a separate component, which refreshes data from the same key, then page data will not get updated.
This is cause any usage of `useAsyncData` created its own instance, and `refresh` function is called within this particular instance. You will see the request in the Network, however this request may not be related to displayed data.
